mm2src/coins/lightning.rs

```diff
@@ -892,7 +905,7 @@ pub async fn open_channel(ctx: MmArc, req: OpenChannelRequest) -> OpenChannelRes
         .save_nodes_addresses(ln_coin.open_channels_nodes)
         .await?;
 
-    ln_coin.persister.add_channel_to_db(pending_channel_details).await?;
+    ln_coin.db.add_channel_to_db(pending_channel_details).await?;
```

Comment:
> can the channel be actually opened without the record in DB?

In the above case, the channel will have no record in the channels history only after the channel is closed. The open channels are saved in memory in the channel manager which gets persisted to the filesystem and loaded again from it on restart, this part of the implementation is on LDK side.

As for having no record in history, there are other instances of saving/updating in DB that this can happen like here for inbound channels https://github.com/KomodoPlatform/atomicDEX-API/blob/577fee5243e1a2468adde1556cec4d1a160795d0/mm2src/coins/lightning/ln_events.rs#L554-L556
And the only solution for all these cases is spawning a loop that retries until successful I guess. Do you think I should do that for all DB operations?

Reply:
> And the only solution for all these cases is spawning a loop that retries until successful I guess. Do you think I should do that for all DB operations?

If such error affects only MM2 internal history of channels, then it's not required, I think. However, returning RPC error in this case might be confusing because the channel will likely be opened anyway.

Also a question: e.g. the node storage is out of space, will channel manager still open a channel if it can't persist its state to the file system?

Reply:
> Also a question: e.g. the node storage is out of space, will channel manager still open a channel if it can't persist its state to the file system?

No, but before persisting the channel to disk, LDK checks if the channel can be opened (accepted by the counterparty) first by sending the `open_channel` msg and waiting for the `accept_channel` msg then within the [watch_channel](https://github.com/lightningdevkit/rust-lightning/blob/0a0f87c00f3c5e30965751fb46502ebc1f110333/lightning/src/chain/chainmonitor.rs#L587) function the channel is persisted [here](https://github.com/lightningdevkit/rust-lightning/blob/0a0f87c00f3c5e30965751fb46502ebc1f110333/lightning/src/chain/chainmonitor.rs#L598), `watch_channel` is used when handling `funding_created` msg by the counterparty [here](https://github.com/lightningdevkit/rust-lightning/blob/0a0f87c00f3c5e30965751fb46502ebc1f110333/lightning/src/ln/channelmanager.rs#L4479), or when the opening node handles `funding_signed` msg [here](https://github.com/lightningdevkit/rust-lightning/blob/0a0f87c00f3c5e30965751fb46502ebc1f110333/lightning/src/ln/channelmanager.rs#L4533). If any of the above fails including the `open_channel` and `accept_channel` messages at the start a `ChannelClosed` event is fired https://github.com/KomodoPlatform/atomicDEX-API/blob/577fee5243e1a2468adde1556cec4d1a160795d0/mm2src/coins/lightning/ln_events.rs#L74-L78
and the failed channel info will be saved in the history DB along with the reason for closure. https://github.com/KomodoPlatform/atomicDEX-API/blob/577fee5243e1a2468adde1556cec4d1a160795d0/mm2src/coins/lightning/ln_events.rs#L397-L399

Reply:
> If such error affects only MM2 internal history of channels, then it's not required, I think. However, returning RPC error in this case might be confusing because the channel will likely be opened anyway.

Do you think we should just log the error in this case?
https://github.com/KomodoPlatform/atomicDEX-API/blob/577fee5243e1a2468adde1556cec4d1a160795d0/mm2src/coins/lightning.rs#L908
This might be moved somewhere else in the future when this feature is implemented https://github.com/lightningdevkit/rust-lightning/issues/1394 since we will have `ChannelOpened`, `ChannelAccepted`, `ChannelConfirmed`events then. I will probably add the channel to DB only on `ChannelConfirmed` or `ChannelClosed` events in this case. But the DB structure will not change, only how we interact with it.

Reply:
> Do you think we should just log the error in this case?

Yes :slightly_smiling_face: